test/fixtures/suit-nesting.css

```diff
@@ -0,0 +1,33 @@
+/** @define Component **/
+
+.Component {
+  color: green;
+
+  .Component-descendant {}
+  &.is-stateName {}
+  &-descendant {}
+  &-descendant--modifier {}
+  &--modifier {}
+  &:pseudo-class {}
+  &:pseudo-element {}
+  &::pseudo-element {}
+  &[attr] {}
+}
+
+.Component-standAloneDescendant {}
+
+.Component {
```

Comment:
A bad pattern perhaps but not invalid as far as bem-linter is concerned. If we think it should be then maybe the SUIT patterns need to be adjusted?